### PR TITLE
improve order fetching and order creation

### PR DESCRIPTION
**Pull request by jhoule86**
- **Temporarily remove `isCancelled` from the PO query due to a Wayfair-side problem involving dryRun**
- change the order creation function's return type from `bool` to `int` so that caller knows ID of plentymarkets order created (if any).
- add general Exception logging for Order Fetching and Order Creation - see https://github.com/wayfair-contribs/plentymarkets-plugin/issues/83
- put the errors from Wayfair during Order Fetching into the Exceptions that we throw
- use different logs for "Order not created" and "Order created but we failed to finish adding data to it," putting the plentymarkets Order's ID in the log details when we know it.
- change "order already exists" logging in Plentymarkets to be at `warning` level instead of `info`.
- make sure we attempt to accept orders that are open and already mapped into Plentymarkets, per @dFed17

### CODE REVIEW

File: src/Core/Api/Services/RegisterPurchaseOrderService.php [hunk 1]
```diff
@@ -26,20 +28,21 @@ class RegisterPurchaseOrderService extends APIService implements RegisterPurchas
    * @return ResponseDTO
    * @throws \Exception
    */
-  public function register(RequestDTO $requestDTO): ResponseDTO {
+  public function register(RequestDTO $requestDTO): ResponseDTO
+  {
 
     $query = 'mutation purchaseOrders { '
```
Comment (mcloughlin77) [inline src/Core/Api/Services/RegisterPurchaseOrderService.php:34]: At some point should we be using a php library for creating graphql queries or are there no good libraries?

Comment (jhoule86) [reply src/Core/Api/Services/RegisterPurchaseOrderService.php:34]: I just took a quick look and I'm not sure how much value such libraries provide. I understand why you're asking, as the "just keep concatenating strings" thing is less-than-graceful, but for the time being I've been treating it as one of the "if it's not broken, don't fix it" situations.

Perhaps @dFed17 has some insight.

File: src/Models/ExternalLogs.php [hunk 2]
```diff
@@ -160,12 +164,17 @@ private function addLogWithMetrics(string $logType, string $message, string $typ
    * @param string $level
    * @param string $message
    */
-  private function addCustomLog(string $level, string $message) {
-    $this->logs[] = [
+  private function addCustomLog(string $level, string $message, $details = null) {
+    $log = [
```
Comment (mcloughlin77) [inline src/Models/ExternalLogs.php:168]: Am I missing something or shouldn't this get added to $logs?

Comment (jhoule86) [reply src/Models/ExternalLogs.php:168]: Good catch. I guess I never finished my thought here.

File: src/Core/Api/Services/RegisterPurchaseOrderService.php (hunk 1 above)
Comment (mcloughlin77) [reply src/Core/Api/Services/RegisterPurchaseOrderService.php:34]: Yeah, I definitely wouldn't recommend changing anything just curious if in the future it was worth using to simply and possible get some additional validation that a library could provide.

**Pull request merged**